engine/src/main/java/com/redhat/cloud/notifications/templates/EmailTemplateMigrationService.java

```diff
@@ -0,0 +1,391 @@
+                "Vulnerability/newCveHighCvssEmailBody", "html", "Vulnerability new CVE high cvss email body"
+        );
+        createInstantEmailTemplate(
+                "rhel", "vulnerability", List.of("new-cve-security-rule"),
+                "Vulnerability/newCveSecurityRuleTitle", "txt", "Vulnerability new CVE security rule email title",
+                "Vulnerability/newCveSecurityRuleBody", "html", "Vulnerability new CVE security rule email body"
+        );
+
+        LOGGER.debug("Migration ended");
+    }
+
+    /*
+     * Creates a template only if it does not already exist in the DB.
+     * Existing templates are never updated by this migration service.
+     */
+    Template getOrCreateTemplate(String name, String extension, String description) {
+        try {
+            Template template = entityManager.createQuery("FROM Template WHERE name = :name", Template.class)
+                    .setParameter("name", name)
+                    .getSingleResult();
+            LOGGER.infof("Template found in DB: %s", name);
+            return template;
+        } catch (NoResultException e) {
+            LOGGER.infof("Creating template: %s", name);
+            Template template = new Template();
+            template.setName(name);
+            template.setDescription(description);
+            template.setData(loadResourceTemplate(name, extension));
+            entityManager.persist(template);
+            return template;
+        }
+    }
+
+    private String loadResourceTemplate(String name, String extension) {
+        String path = "/templates/" + name + "." + extension;
+        try (InputStream inputStream = getClass().getClassLoader().getResourceAsStream(path)) {
+            if (inputStream == null) {
+                throw new IllegalStateException("Template resource file not found: " + path);
+            } else {
+                return new String(inputStream.readAllBytes(), UTF_8);
+            }
+        } catch (IOException e) {
+            throw new UncheckedIOException("Template loading from resource file failed: " + path, e);
+        }
+    }
+
+    /*
+     * Creates an instant email template and the underlying templates only if:
+     * - the event type exists in the DB
+     * - the instant email template does not already exist in the DB
+     * Existing instant email templates are never updated by this migration service.
+     */
+    private void createInstantEmailTemplate(String bundleName, String appName, List<String> eventTypeNames,
+            String subjectTemplateName, String subjectTemplateExtension, String subjectTemplateDescription,
+            String bodyTemplateName, String bodyTemplateExtension, String bodyTemplateDescription) {
+
+        for (String eventTypeName : eventTypeNames) {
+            Optional<EventType> eventType = findEventType(bundleName, appName, eventTypeName);
+            if (eventType.isPresent() && !instantEmailTemplateExists(eventType.get())) {
```

Comment:
We can't throw an exception because the event types may vary between stage and prod.

I will add more log entries and a migration report that lists all the anomalies and print that in the log as well.

Reply:
Sounds good

Reply:
I pushed a new commit. The REST endpoint now returns all warnings that were raised during the migration. That includes two additional warnings you suggested above when the event type or the app is not found in the DB.